wcfsetup/install/files/lib/system/search/IContextAwareSearchEngine.class.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace wcf\system\search;
+
+use wcf\system\database\util\PreparedStatementConditionBuilder;
+
+/**
+ * Default interface for search engines that support
+ * filtering by context.
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Search
+ * @since 6.0
+ */
+interface IContextAwareSearchEngine extends ISearchEngine
+{
+    /**
+     * Returns the condition builder class name required to provide conditions for getInnerJoin().
+     *
+     * @return  string
+     */
+    public function getConditionBuilderClassName();
+
+    /**
+     * Returns the inner join query and the condition parameters. This method is allowed to return NULL for both the
+     * 'fulltextCondition' and 'searchIndexCondition' index instead of a PreparedStatementConditionBuilder instance:
+     *
+     * array(
+     *  'fulltextCondition' => $fulltextCondition || null,
+     *  'searchIndexCondition' => $searchIndexCondition || null,
+     *  'sql' => $sql
+     * );
```

Comment:
```suggestion
     * @return array{
     *             fulltextCondition: ?whateverthetypeshouldbe
     *             searchIndexCondition: ?whateverthetypeshouldbe
     *             sql: string
     *         }
```

This uses the array shape syntax as supported by Psalm: https://psalm.dev/docs/annotating_code/type_syntax/array_types/#array-shapes

Reply:
This was copied from `getInnerJoin()` and the type is `?PreparedStatementConditionBuilder`. However, my IDE doesn’t support this syntax and I have no other tool at hand to validate it. Please feel free to add the proper typing to this interface and the base interface.